graph-commons/src/main/scala/io/renku/crypto/AesCrypto.scala

```diff
@@ -63,31 +64,40 @@ abstract class AesCrypto[F[_]: MonadThrow, NONENCRYPTED, ENCRYPTED](secret: Secr
 object AesCrypto {
 
   final class Secret private (val value: ByteVector) extends AnyVal {
-    override def toString: String = "<sensitive>"
-    def toBase64:          String = value.toBase64
+    override def toString: String                                   = "<sensitive>"
+    def decodeAscii:       Either[CharacterCodingException, String] = value.decodeAscii
   }
 
   object Secret {
 
     def apply(value: ByteVector): Either[String, Secret] = {
       val expectedLengths = List(16L, 24L, 32L)
-      if (expectedLengths.contains(value.length)) Right(new Secret(value))
+      if (expectedLengths contains value.length) Right(new Secret(value))
       else Left(s"Expected ${expectedLengths.mkString(" or ")} bytes, but got ${value.length}")
     }
 
     def unsafe(value: ByteVector): Secret = apply(value).fold(sys.error, identity)
 
-    def fromBase64(b64: String): Either[String, Secret] =
+    private def fromBase64(b64: String): Either[String, Secret] =
       ByteVector.fromBase64Descriptive(b64).flatMap(apply)
 
     def unsafeFromBase64(b64: String): Secret =
       fromBase64(b64).fold(sys.error, identity)
 
     implicit def secretReader: ConfigReader[Secret] =
-      base64ByteVectorReader.emap(bv =>
-        Secret(bv.takeWhile(_ != 10)).leftMap(err =>
+      base64SecretReader orElse asciiSecretReader
```

Comment:
Does this mean, the ascii secret must not be valid base64? This could happen in theory, perhaps?

Reply:
Yes, I know what you mean. It can happen and yes, if it does it's quite unlikely it will be 16, 24 or 32 chars long after decoding. We'll simply fail horrendously. To be honest I don't really see an easy solution for that assuming we'd like to have a single secret entry in the helm config. From the other side, I guess our yat will be the only people who generate the values (in an automatic way), so perhaps that's acceptable. What do you think? I can talk to Alessandro and check if they are happy with that.

Reply:
I'm fine of course. Just worth noting, I guess. If it occurs, it will be something hard to find in a year 😄 And might be a support case for deployments not on our control. But agree that it is not very likely. And if we are limited by helm or whatever other tool, I think there is not much we can do.